Added Notification methods to Service Worker API

SteffenJahr opened the pull request:
Please fill in this template.

- [X] Make your PR against the `master` branch.
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [X] Run `tsc` without errors.
- [X] Run `npm run lint package-name` if a `tslint.json` is present. 
=> **My code doesn't raise any lint conditions but there were "errors" before that I can't fix: `void-return  Use the 'void' type for return types only. Otherwise, use 'undefined'.`**

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/showNotification and https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/getNotifications
- [-] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

SteffenJahr commented:
@tristancaron Can you please review :) dt-bot seems not to work here :)

DanielRosenwasser commented on service_worker_api/index.d.ts:
```diff
@@ -576,12 +576,74 @@ interface ServiceWorkerRegistration extends EventTarget {
+    requireInteraction: boolean;
+    tag: string;
+    vibrate: number[];
+    data: any;
```
Should these all be optional?

SteffenJahr replied:
Thank you for your feedback. I have updated the PR.

SteffenJahr replied:
Can you please review again @DanielRosenwasser :)

SteffenJahr commented:
Can you please review again @DanielRosenwasser :) I have updated the ServiceWorkerNotificationOptions to make all properties optional.

aozgaa commented on service_worker_api/index.d.ts:
```diff
@@ -576,12 +576,73 @@ interface ServiceWorkerRegistration extends EventTarget {
+     * Displays the notification with the requested title.
+     * @param [title] The title that must be shown within the notification
+     * @param [options] An object that allows to configure the notification. It can have the following properties:
+     * actions: An array of actions to display in the notification. The members of the array should be an object
```
Can you transfer the descriptions of the properties to their declarations in the `ServiceWorkerNotificationOptions` interface?

aozgaa commented on service_worker_api/service_worker_api-tests.ts:
```diff
@@ -191,3 +191,16 @@ self.addEventListener('notificationclick', function(event: NotificationEvent) {
 });
+
+navigator.serviceWorker.ready.then(function(registration) {
+    registration.showNotification('Notification Sample', <ServiceWorkerNotificationOptions>{
```
Since the properties are optional, do you need the type assertion?

aozgaa commented on service_worker_api/index.d.ts:
```diff
@@ -576,12 +576,73 @@ interface ServiceWorkerRegistration extends EventTarget {
+    actions?: string[];
+    badge?: string;
+    body?: string;
+    dir?: string;
```
Can you make the type `'auto' | 'ltr' | 'rtl'`?

SteffenJahr commented:
@aozgaa Thank you for your feedback. I have updated the commit and also amended the options a little bit. I also updated the tests and removed the type assertion because you are right and I don't need it.

zhengbli commented:
@DanielRosenwasser further comments on this one?

SteffenJahr commented:
@zhengbli @DanielRosenwasser Any news here?
